xmtp_mls/src/storage/encrypted_store/mod.rs

```diff
@@ -243,6 +244,29 @@ macro_rules! impl_store {
     };
 }
 
+// Inserts the model into the database by primary key, silently skipping on unique constraints
+#[macro_export]
+macro_rules! impl_store_or_ignore {
+    ($model:ty, $table:ident) => {
+        impl
+            $crate::StoreOrIgnore<$crate::storage::encrypted_store::db_connection::DbConnection<'_>>
+            for $model
+        {
+            fn store_or_ignore(
+                &self,
+                into: &$crate::storage::encrypted_store::db_connection::DbConnection<'_>,
+            ) -> Result<(), $crate::StorageError> {
+                let result = into.raw_query(|conn| {
+                    diesel::insert_into($table::table)
+                        .values(self)
+                        .execute(conn)
+                });
+                $crate::storage::ignore_unique_violation(result)
```

Comment:
Note - specifically ignoring unique constraint violations. Other approaches (for example, sqlite's INSERT OR IGNORE, ON CONFLICT, etc) will throw away non-null constraint errors, which risk us losing DB writes without realizing it. There are other places in the codebase where we do this, not sure if those need to be changed.

Reply:
That's a very good point. We should fix those other call sites as well

Reply:
Will fix in next PR